graphx/src/main/scala/org/apache/spark/graphx/lib/PageRank.scala

```diff
@@ -189,6 +189,35 @@ object PageRank extends Logging {
       numIter: Int,
       resetProb: Double = 0.15,
       sources: Array[VertexId]): Graph[Vector, Double] = {
+    val sourcesWithScores = sources.map((_, 1.0))
+    runParallelPersonalizedPageRank(graph, numIter, resetProb, sourcesWithScores)
+  }
+
+  /**
+   * Run Personalized PageRank for a fixed number of iterations, for a
+   * set of starting nodes in parallel. Returns a graph with vertex attributes
+   * containing the pagerank relative to all starting nodes (as a sparse vector) and
+   * edge attributes the normalized edge weight
+   *
+   * @tparam VD The original vertex attribute (not used)
+   * @tparam ED The original edge attribute (not used)
+   *
+   * @param graph The graph on which to compute personalized pagerank
+   * @param numIter The number of iterations to run
+   * @param resetProb The random reset probability
+   * @param sources The list of (source, initial score) to compute personalized pagerank from
+   * @return the graph with vertex attributes
+   *         containing the pagerank relative to all starting nodes (as a sparse vector
+   *         indexed by the position of nodes in the sources list) and
+   *         edge attributes the normalized edge weight
+   *
+   * @since 3.0.0
```

Comment:
Oh, this has to be an annotation like `@Since("3.0.0")` on the method, not within the scaladoc

Reply:
I got confused by seeing this. https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/SQLImplicits.scala
I fixed it now.